sw/airborne/modules/gps/gps_ubx_ucenter.c

```diff
@@ -258,56 +272,62 @@ static bool gps_ubx_ucenter_autobaud(uint8_t nr)
     case 2:
       gps_ubx_ucenter.reply = GPS_UBX_UCENTER_REPLY_NONE;
       uart_periph_set_baudrate(&(UBX_GPS_LINK), B38400); // Try the most common first?
+      gps_ubx_ucenter.cnt_valid = 0;
       gps_ubx_ucenter_config_port_poll();
       break;
     case 3:
-      if (gps_ubx_ucenter.reply == GPS_UBX_UCENTER_REPLY_ACK) {
-        gps_ubx_ucenter.baud_init = gps_ubx_ucenter.baud_run;
+      if (gps_ubx_ucenter.cnt_valid > GPS_UBX_UCENTER_MIN_VALID || gps_ubx_ucenter.reply == GPS_UBX_UCENTER_REPLY_ACK) {
+        gps_ubx_ucenter.baud_init = 38400;
```

Comment:
so the `baud_run` set by a reply to gps_ubx_ucenter_config_port_poll() is not used anymore at all?